code/game/machinery/_machinery.dm

```diff
@@ -531,7 +531,8 @@ Class Procs:
 /obj/machinery/tesla_act(power, tesla_flags, shocked_objects)
 	..()
 	if(prob(85) && (tesla_flags & TESLA_MACHINE_EXPLOSIVE))
-		explosion(src, 1, 2, 4, flame_range = 2, adminlog = FALSE, smoke = FALSE)
+		if(!(resistance_flags & INDESTRUCTIBLE))
```

Comment:
why have the second if when you could just put that condition in the if its nested inside?

Reply:
fixed